<?xml version="1.0" encoding="utf-8"?>
<!--
Parameters for Center for robotics summer school 2019
-->

<launch>

	<arg name="map_frame" default="/map"/>

	<node name="mapper" type="dynamic_mapper_node" pkg="ethzasl_icp_mapper" output="screen" >
		<remap from="cloud_in" to="/rslidar_points" />
		<param name="subscribe_scan" value="false"/>
		<param name="subscribe_map" value="false"/>
		<param name="icpConfig"
			   value="$(find ethzasl_icp_mapper)/launch/supermegabot/icp_dynamic.yaml" />
		<param name="inputFiltersConfig"
			   value="$(find ethzasl_icp_mapper)/launch/supermegabot/input_filters.yaml" />
		
		<!--Put that back! -->
		<param name="publishMapTf" value="false"/>
		<param name="odom_frame" value="/realsense_t265_odom_frame" />
		<param name="base_frame" value="/base" />
		<param name="lidar_frame" value="/rslidar" />
		<param name="map_frame" value="$(arg map_frame)" />
		<param name="useROSLogger" value="true" />
		<param name="inputQueueSize" value="1" />
		<param name="tfRefreshPeriod" value="0.005" />
		<param name="minOverlap" value="0.2" />
		<param name="maxOverlapToMerge" value="0.96" /> 
		<param name="minReadingPointCount" value="1000" /> 
		<param name="minMapPointCount" value="10000" /> <!-- quick fix to have map published each time -->
		<param name="skip_frames" value="2" /> <!-- Skip this many frames before processing a new one-->
		<param name="publish_rate" value="100" />
	</node>

	<node pkg="tf" type="static_transform_publisher" name="tracking_cam_to_base"
				args="-0.309 -0.008 -0.351 0 0 0 1  /realsense_t265_pose_frame /base 100"/>
</launch>
